modules/openid-federation-common/src/commonMain/kotlin/com/sphereon/oid/fed/common/httpclient/OidFederationClient.kt

```diff
@@ -0,0 +1,69 @@
+package com.sphereon.oid.fed.common.httpclient
+
+import com.sphereon.oid.fed.openapi.models.EntityStatement
+import io.ktor.client.*
+import io.ktor.client.call.*
+import io.ktor.client.engine.*
+import io.ktor.client.plugins.auth.*
+import io.ktor.client.plugins.auth.providers.*
+import io.ktor.client.plugins.cache.*
+import io.ktor.client.plugins.contentnegotiation.*
+import io.ktor.client.plugins.logging.*
+import io.ktor.client.request.*
+import io.ktor.client.request.forms.*
+import io.ktor.http.*
+import io.ktor.http.HttpMethod.Companion.Get
+import io.ktor.http.HttpMethod.Companion.Post
+import io.ktor.serialization.kotlinx.json.*
+import io.ktor.utils.io.core.*
+
+class OidFederationClient(
+    engine: HttpClientEngine,
+    private val isRequestAuthenticated: Boolean = false,
+    private val isRequestCached: Boolean = false
+) {
+    private val client: HttpClient = HttpClient(engine) {
+        install(HttpCache)
+        install(ContentNegotiation) {
+            register(EntityStatementJwt, EntityStatementJwtConverter())
+            json()
+        }
+        install(Logging) {
+            logger = Logger.DEFAULT
+            level = LogLevel.INFO
+        }
+        if (isRequestAuthenticated) {
+            install(Auth) {
+                bearer {
+                    loadTokens {
+                        //TODO add correct implementation later
+                        BearerTokens("accessToken", "refreshToken")
+                    }
+                }
+            }
+        }
+        if (isRequestCached) {
+            install(HttpCache)
+        }
+    }
+
+    suspend fun fetchEntityStatement(url: String, httpMethod: HttpMethod = Get, parameters: Parameters = Parameters.Empty): EntityStatement {
```

Comment:
I think all three methods below can be replaced with just

```
    suspend fun fetchEntityStatement(
        url: String,
        httpMethod: HttpMethod = HttpMethod.Get,
        parameters: Parameters = Parameters.Empty
    ): EntityStatement = client.request(url) {
        this.method = httpMethod
        if (httpMethod == HttpMethod.Post) {
            setBody(FormDataContent(parameters))
        }
    }.body()
```

Reply:
Please don't. More work is needed in the future on the post side. So having seperate methods for these is preferable

Reply:
I've like the suggestion and it works. Code updated

Reply:
As per Niels's comment revert
```
    suspend fun fetchEntityStatement(url: String, httpMethod: HttpMethod = HttpMethod.Get, parameters: Parameters = Parameters.Empty): EntityStatement {
        return when (httpMethod) {
            HttpMethod.Get -> getEntityStatement(url)
            HttpMethod.Post -> postEntityStatement(url, parameters)
            else -> throw IllegalArgumentException("Unsupported HTTP method: $httpMethod")
        }
    }

    private suspend fun getEntityStatement(url: String): EntityStatement =
        client.get(url).body()

    private suspend fun postEntityStatement(url: String, parameters: Parameters): EntityStatement =
        client.post(url) {
            setBody(FormDataContent(parameters))
        }.body()
```

Reply:
I've got it, the Github UI was not updated

Reply:
Yes Niels's comment came in seconds before your post

Reply:
reverted